### DISCUSSION TITLE
Handling/Organizing COSMOS DB Access & Service Bus within this Architecture

### DISCUSSION THREAD

**Discussion by sforsandeep** [jasontaylordev/CleanArchitecture#615] [Q&A]
Hi,

I really love this architecture and being a team lead I was able to recommend and convince my developers to use this. 

But I really need a support  now where we got stuck. Ours is a Function App with which we are using Azure COSMOS DB and Service Bus. 

Kindly note, that it's a single solution (But with multiple Azure Function Apps projects sharing the common Domain, Application & Infrastructure). We made it a separate function App project instead of separate functions within a single Function App because the client expects each of the Function Apps needs to be handled as separate so they can even do the individual scaling when required.

When I tell something about the COSMOS side, it will be a huge Single DB, but with minimal containers, each holding a huge set of data.

So we got stuck organizing this COSMOS DB handling within this architecture. Requesting your help.

Next is the Service bus, we need to do lots of in/out of service bus messages as well. Here as well not sure how to well organize its handling. If someone can guide how to make it in this, that would be more than just helpful and we can follow the same for other DBs and other services coming in the future.

Please help

**Comment by bramburn** [comment]
you just need to configure the Dependencyinjection file which ahs the applicationdbcontext injection into the service to configure it. I thought using EF and dbcontext would allow you to configure it?

**Reply by bramburn** [reply]
what you could also do is separate the app and the tables in the database for each domain, so marketing, sales, projects, etc... would each be their own subdomain and you open up API to allow them to communicate between them. use the CQRS or a MQ to send data across to multiple subdomains for tasks.
